File: frontend/src/objects/order/AfterSalesRequestType.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
export const AFTER_SALES_REQUEST_TYPE = {
  returnRequest: 'ReturnRequest',
  compensationRequest: 'CompensationRequest',
} as const

export type AfterSalesRequestType =
  (typeof AFTER_SALES_REQUEST_TYPE)[keyof typeof AFTER_SALES_REQUEST_TYPE]

File: backend/src/main/scala/services/order/objects/AfterSalesRequestType.scala
package services.order.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given
import system.objects.*

import io.circe.{Decoder, Encoder}

enum AfterSalesRequestType:
  case ReturnRequest, CompensationRequest

object AfterSalesRequestType:
  private val enumLabel = text("AfterSalesRequestType")
  def render(value: AfterSalesRequestType): DisplayText = enumDisplayText(value)
  def parse(value: DisplayText): Option[AfterSalesRequestType] = parseEnumValue(value, AfterSalesRequestType.values)
  given Encoder[AfterSalesRequestType] = enumEncoder
  given Decoder[AfterSalesRequestType] = enumDecoder(AfterSalesRequestType.values, enumLabel)
